### ISSUE TITLE
Support Hierarchical Controlled Vocabularies

### ISSUE THREAD

**Issue by eaquigley** [IQSS/dataverse#434]
---

Author Name: **Eleni Castro** (@posixeleni)
Original Redmine Issue: 3849, https://redmine.hmdc.harvard.edu/issues/3849
Original Date: 2014-04-12
Original Assignee: Gustavo Durand

---

We will need to be able to support lists of hierarchical controlled vocabularies. We currently only support flat lists. This will be important for biomedical metadata and for GSD metadata block.

For example:
- Semester (Fall 2013)
  *\* Course Name (only list of courses for Fall 2013)
  *\* Faculty Name (only list of faculty for Fall 2013)

**Comment by bencomp** [comment]
This sounds like the vocabulary has several facets; three fields could be used for this case. I don't think faculty teaching in Fall 2013 is conceptually different from faculty teaching in other semesters?

However, controlled vocabularies, especially subject vocabularies or geographical coverage vocabularies, are rarely flat lists. Having identifiers for terms in a vocabulary can help structure them (see #947). But (althought I'm not assigned) I'm not sure what kind of support is envisioned here.

**Comment by bencomp** [comment]
Giving it another thought: one way of support for hierarchical controlled vocabularies is including all dataverses and datasets that have been assigned vocabulary term `1.X` when in a facet only term `1` is selected (and `1.X` is a narrower term than `1`).

**Comment by posixeleni** [comment]
@bencomp you're right the example above wasnt a good one. A better example is the NCBI Taxonomy for Organisms used in Life Sciences studies: http://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?name=Eubacteria

![image](https://cloud.githubusercontent.com/assets/5013837/6600875/5c9cc0c8-c7cf-11e4-836e-d279f75fe4e8.png)

**Comment by pdurbin** [comment]
This is something @bmckinney at least mentioned last week so I added the SBGrid label.

**Comment by mheppler** [comment]
With the potential to require this feature in association with a custom Harvard Dataverse homepage (#2003), I have added the 4.7 milestone label.

**Comment by pdurbin** [comment]
There is discussion about the subject field in #1815 that should be reviewed. These issues seem quite similar to me.

**Comment by pdurbin** [comment]
> This will be important for biomedical metadata

Related:

- #8681

These days we support external vocabularies but I'm not sure if any are hierarchical (I've only played with ORCID and ROR): https://guides.dataverse.org/en/latest/admin/metadatacustomization.html#using-external-vocabulary-services

**Comment by cmbz** [comment]
2025/01/13: Closing because there is another open issue that will address hierarchical controlled vocabs: https://github.com/IQSS/dataverse-pm/issues/236
